File: HashTable/383_Ransom_Note/383_Ransom_Note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char, int> r;
        unordered_map<char, int> m;

        for(auto& c: ransomNote)
            ++r[c];

        for(auto& c: magazine)
            ++m[c];

        for (auto& it: r) {
            if (m.find(it.first) == m.end())
                return false;
            if(m[it.first] < it.second)
                return false;
        }
        return true;
    }
};

File: HashTable/383_Ransom_Note/383_Ransom_Note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        HashMap<Character, Integer> r = new HashMap<>();
        HashMap<Character, Integer> m = new HashMap<>();

        for(Character c : ransomNote.toCharArray())
            r.merge(c, 1, Integer::sum); 

        for(Character c : magazine.toCharArray())
            m.merge(c, 1, Integer::sum); 

        for(Map.Entry<Character, Integer> e : r.entrySet()) {
            char key = e.getKey();    
            Integer value = e.getValue(); 
            if(!m.containsKey(key))
                return false;
            if(m.get(key)<value)
                return false;

        }
        return true;        
    }
}